sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageBlobContainersTests.java

```diff
@@ -54,18 +51,23 @@ public void canCreateBlobContainer() {
             .define(saName)
             .withRegion(Region.US_EAST)
             .withNewResourceGroup(rgName)
+            .withSku(StorageAccountSkuType.STANDARD_LRS)
+            .disableSharedKeyAccess()
+            // It's not taking effect during storage account creation. Not sure if it's a bug.
+            //            .enableBlobPublicAccess()
             .create();
 
         BlobContainers blobContainers = this.storageManager.blobContainers();
         BlobContainer blobContainer = blobContainers.defineContainer("blob-test")
             .withExistingStorageAccount(rgName, saName)
-            .withPublicAccess(PublicAccess.CONTAINER)
+            .withPublicAccess(PublicAccess.NONE)
+            //            .withPublicAccess(PublicAccess.CONTAINER)
```

Comment:
We need StorageAccount level `allowBlobPublicAccess` be enabled for container/blobs to be able to be public accessible. Otherwise, service will return `Public access is not allowed for this storage account`.